command/build_test.go

```diff
@@ -832,7 +831,7 @@ func TestHCL2PostProcessorForceFlag(t *testing.T) {
 
 	// Second build should override previous manifest
 	UUID, _ = uuid.GenerateUUID()
-	os.Setenv("PACKER_RUN_UUID", UUID)
+	t.Setenv("PACKER_RUN_UUID", UUID)
```

Comment:
> I wonder about this test, what's the behaviour of `t.Setenv`? Does it unset the variable, or does it reset its value after the test? If the latter is true, which value will that set when two occurrences are called? Will that be the value before the test, or the value before the second call?

@lbajolet-hashicorp  From `go/src/testing/testing.go`, the cleanup from second `t.Setenv` will reset the variable value, but not unset the variable because `"PACKER_RUN_UUID"` already presents in the environment.
```go
// Setenv calls os.Setenv(key, value) and uses Cleanup to
// restore the environment variable to its original value
// after the test.
//
// This cannot be used in parallel tests.
func (c *common) Setenv(key, value string) {
	c.checkFuzzFn("Setenv")
	prevValue, ok := os.LookupEnv(key)

	if err := os.Setenv(key, value); err != nil {
		c.Fatalf("cannot set environment variable: %v", err)
	}

	if ok {
		c.Cleanup(func() {
			os.Setenv(key, prevValue)
		})
	} else {
		c.Cleanup(func() {
			os.Unsetenv(key)
		})
	}
}
```

`Cleanup` functions are called in last added, first called order (https://pkg.go.dev/testing#T.Cleanup). So for example, the **second** `t.Setenv` cleanup will reset the value back to `ABC`, but the **first** `t.Setenv` cleanup will unset the environment variable because `"PACKER_RUN_UUID"` doesn't exist in the environment at the very beginning.
```go
func TestHCL2PostProcessorForceFlag(t *testing.T) {
	t.Helper()

	UUID, _ := uuid.GenerateUUID() // assume it's "ABC"
	// Manifest will only clean with force if the build's PACKER_RUN_UUID are different
	t.Setenv("PACKER_RUN_UUID", UUID)

	...

	// Second build should override previous manifest
	UUID, _ = uuid.GenerateUUID() // assume it's "XYZ"
	t.Setenv("PACKER_RUN_UUID", UUID)
}
```

Reply:
Hey @Juneezee,
Sorry I took long before revisiting, that slipped my mind.

Sounds good to me then, we can merge this!